graphstorm-processing/scripts/submit_gsp_emr_step.py

```diff
@@ -0,0 +1,245 @@
+            f"{gsp_ecr_registry}/{ecr_repo_name}:{image_tag}"
+        ),
+        "--conf",
+        "spark.yarn.appMasterEnv.YARN_CONTAINER_RUNTIME_TYPE=docker",
+        "--conf",
+        (
+            "spark.yarn.appMasterEnv.YARN_CONTAINER_RUNTIME_DOCKER_IMAGE="
+            f"{gsp_ecr_registry}/{ecr_repo_name}:{image_tag}"
+        ),
+        "--conf",
+        "spark.executorEnv.PYSPARK_PYTHON=/.pyenv/shims/python",
+        "--conf",
+        "spark.yarn.appMasterEnv.PYSPARK_PYTHON=/.pyenv/shims/python",
+        s3_entry_point,
+    ] + args.gsp_arguments.split()
+
+    # Define cluster configuration
+    cluster_config = {
+        "Name": f"gsp-emr-{core_instance_count}x-{instance_type}",
+        "ReleaseLabel": args.emr_version,
+        "Applications": [{"Name": "Hadoop"}, {"Name": "Spark"}],
+        "Configurations": [
+            {
+                "Classification": "container-executor",
+                "Configurations": [
+                    {
+                        "Classification": "docker",
+                        "Properties": {
+                            "docker.privileged-containers.registries": f"local,centos,{gsp_ecr_registry}",  # pylint: disable=line-too-long
+                            "docker.trusted.registries": f"local,centos,{gsp_ecr_registry}",
+                        },
+                    }
+                ],
+                "Properties": {},
+            },
+            {
+                "Classification": "spark",
+                "Properties": {"maximizeResourceAllocation": "true"},
+            },
+        ],
+        "ServiceRole": args.service_role,
+        "JobFlowRole": args.jobflow_role,
+        "VisibleToAllUsers": True,
+        "AutoTerminationPolicy": {"IdleTimeout": timeout_hours * 60 * 60},
+        "Instances": {
+            "InstanceGroups": [
+                {
+                    "Name": "Master node",
+                    "Market": "ON_DEMAND",
+                    "InstanceRole": "MASTER",
+                    "InstanceType": instance_type,
+                    "InstanceCount": 1,
+                },
+                {
+                    "Name": "Core nodes",
+                    "Market": "ON_DEMAND",
+                    "InstanceRole": "CORE",
+                    "InstanceType": instance_type,
+                    "InstanceCount": core_instance_count,
+                },
+            ],
+            "KeepJobFlowAliveWhenNoSteps": False,
+        },
+        "ScaleDownBehavior": "TERMINATE_AT_TASK_COMPLETION",
+        "Steps": [
+            {
+                "Name": "GSProcessing",
+                "ActionOnFailure": "TERMINATE_CLUSTER",
+                "HadoopJarStep": {"Jar": "command-runner.jar", "Args": task_arguments},
+            }
+        ],
+        "Tags": [{"Key": "Launching user", "Value": f"{os.environ.get('USER', 'None')}"}],
+    }
+
+    print("Submitted task:")
+    pprint(task_arguments)
```

Comment:
typo?

Reply:
Wee use the p(retty)print lib to print the args in a nice format. See https://docs.python.org/3/library/pprint.html